internal/updater/controller.go

```diff
@@ -0,0 +1,105 @@
+package updater
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/quay/claircore/internal/vulnstore"
+	"github.com/quay/claircore/libvuln/driver"
+)
+
+// Controller is the interface that updater Controllers implement.
+type Controller interface {
+	// Run has the Controller execute all the Updaters passed on the channel,
+	// until it's closed. The method runs synchronously and only returns after
+	// the channel is closed or the context is canceled.
+	//
+	// Any spawned goroutines should inherit the passed-in Context.
+	//
+	// A call to Run should be thought of as one execution of the Updaters; that
+	// is, the caller should not use a single channel if it's looping, it should
+	// instead call Run on each iteration of the loop.
```

Comment:
"it should instead call Run with a new channel on each iteration of the loop" Is that the point you are trying to convey?